.github/workflows/build-alpha.yml

```
@@ -10,47 +10,67 @@ on:
         -   pyscriptjs/**
         -   .github/workflows/build-alpha.yml # Test that workflow works when changed
 
+env:
+    MINICONDA_PYTHON_VERSION: py38
+    MINICONDA_VERSION: 4.11.0
+
+defaults:
+    run:
+        working-directory: pyscriptjs
+
 jobs:
     build:
         runs-on: ubuntu-latest
         permissions:
             contents: read
             id-token: write
-        defaults:
-            run:
-                working-directory: pyscriptjs
-
         steps:
+
         -   name: Checkout
             uses: actions/checkout@v3
+
         -   name: Install node
             uses: actions/setup-node@v3
             with:
                 node-version: 18.x
+
         -   name: Cache node modules
             uses: actions/cache@v3
             env:
                 cache-name: cache-node-modules
             with:
-                # npm cache files are stored in `~/.npm` on Linux/macOS
+              # npm cache files are stored in `~/.npm` on Linux/macOS
                 path: ~/.npm
                 key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
                 restore-keys: |
                     ${{ runner.os }}-build-${{ env.cache-name }}-
                     ${{ runner.os }}-build-
                     ${{ runner.os }}-
+
+        -   name: Install miniconda
+            run: |
+                curl -L -O https://repo.continuum.io/miniconda/Miniconda3-${MINICONDA_PYTHON_VERSION}_${MINICONDA_VERSION}-Linux-x86_64.sh \
+                && bash Miniconda3-${MINICONDA_PYTHON_VERSION}_${MINICONDA_VERSION}-Linux-x86_64.sh -b -p conda/
```

It may make sense to use setup-miniconda here since it abstracts this away and falls back to the [already installed conda](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md), making the whole process much faster.

That said this should probably use the right URL if setup-miniconda isn't something you'd like to use:
```suggestion
        -   name: Install miniconda
            run: |
                curl -L -O https://repo.anaconda.com/miniconda/Miniconda3-${MINICONDA_PYTHON_VERSION}_${MINICONDA_VERSION}-Linux-x86_64.sh \
                && bash Miniconda3-${MINICONDA_PYTHON_VERSION}_${MINICONDA_VERSION}-Linux-x86_64.sh -b -p conda/
```

---

Oh fascinating it didn't occur to me we'd be built into the cloud runners already! Thanks for looking out, that does sound a lot easier.

As you can probably tell I don't have enough experience with conda directly so this is great stuff to learn. I assume that action will add it to PATH? It was a bit of juggling to execute it in the CI without changing the Makefile too much.

---

> Oh fascinating it didn't occur to me we'd be built into the cloud runners already! Thanks for looking out, that does sound a lot easier.

Yep, agreed!

> As you can probably tell I don't have enough experience with conda directly so this is great stuff to learn. I assume that action will add it to PATH? It was a bit of juggling to execute it in the CI without changing the Makefile too much.

When using the setup-miniconda action it's added to PATH, but GitHub doesn't put it on PATH automatically on its own to not accidentally override other built-in binaries.